### PR TITLE
Enable cypher acceptance testing on the openCypher TCK infrastructure

### PR DESCRIPTION
**Pull request by boggle**
This PR changes the module structure of community/cypher to enable writing acceptance tests in the same style as openCypher TCK tests.  This is done to align how we write acceptance tests with the requirements of openCypher and - going forward - to make it easy to move features to the openCypher TCK.

In particular, this PR adds three new modules
- `compatibility-spec-suite` (former `compatibility-suite` module)
- `acceptance-spec-suite` (former `acceptance` module)
- `spec-suite-tools` (shared code between both)

Additionally, this PR ports over the majority of `ShortestPathAcceptanceTest` to a new feature in `acceptance-spec-suite`

### CODE REVIEW

**Comment by boggle** [comment]
Pending merge of https://github.com/opencypher/openCypher/pull/70

**Comment by Mats-SX** [comment]
@boggle Given the latest discussions, this PR ought to be simply closed.

**Comment by boggle** [comment]
@Mats-SX Propably yes, though it would be nice to salvage the feature file somehow. Let's discuss once I'm back.

File: community/cypher/cypher-compiler-3.1/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
   </scm>
 
   <properties>
-    <version-package>cypher.internal.compiler.v3_0</version-package>
+    <version-package>cypher.internal.compiler.v3_1</version-package>
```
Comment (systay) [inline community/cypher/cypher-compiler-3.1/pom.xml:25]: What are these used for? I don't see any uses of this property in this file

Comment (Mats-SX) [inline community/cypher/cypher-compiler-3.1/pom.xml:25]: It seems to come from https://github.com/neo4j/neo4j/commit/219fbf033de8914f1fefc5e362a694778f421393#diff-51794d1cb1dcb3ed90452bd7f0707e25R24, but perhaps this is no longer used.

**Pull request merged**